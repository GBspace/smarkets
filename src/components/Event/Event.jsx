
import React from 'react'
import {Link} from 'react-router-dom';
const Event = ({id}) => {
  return (
    <div>
        <Link to={`/event/${id}`} >
            <span>{id}</span>
        </Link>
    </div>
  )
}

export default Event