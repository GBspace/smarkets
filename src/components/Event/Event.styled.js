import { styled } from "@mui/system";

export const StyledEvent = styled('div')(({theme})=>({
    a: {
        textDecoration: 'none',
        color: theme.palette.primary.contrastText,
        border: '2px',
    },
    transition:`${theme.transitions.create(['background-color', 'transform'], {
        duration: theme.transitions.duration.standard,
      })}`,
}));