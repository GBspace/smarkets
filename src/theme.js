import {createTheme} from '@mui/material/styles';

// A custom theme for this app
const customTheme = createTheme({
    palette: {
      type: 'light',
      primary: {
        main: '#696969',
        contrastText: '#282c34'
      },
    },
    overrides: {
      MuiPaper: {
        root: {
          padding: '20px 10px',
          margin: '10px',
          backgroundColor: '#fff',
        },
      },
      MuiButton: {
        root: {
          margin: '5px',
        },
      },
    },
  });
  export {customTheme};