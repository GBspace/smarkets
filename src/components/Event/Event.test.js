import Event from './Event';
import {shallow} from 'enzyme';
import * as React from "react";
import {customTheme} from '../../theme';
import toJson from 'enzyme-to-json';

describe('EventComponent', ()=>{
    it('renders correctly', () => {
        const wrapper = shallow(<Event id={1} theme={customTheme}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
      });

    it('should trigger the router link on being clicked',()=>{
        const wrapper = shallow(<Event id={1}/>);
        const link = wrapper.find('Link');
        expect(link.at(0).props().to).toBe('/event/1');
    });
})