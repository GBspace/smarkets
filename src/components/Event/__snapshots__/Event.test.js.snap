// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventComponent renders correctly 1`] = `
<Styled(div)>
  <Link
    to="/event/1"
  >
    <WithStyles(ForwardRef(ListItem))>
      <ForwardRef(ListItemButton)>
        <WithStyles(ForwardRef(ListItemText))
          primary={1}
        />
      </ForwardRef(ListItemButton)>
    </WithStyles(ForwardRef(ListItem))>
  </Link>
</Styled(div)>
`;
